### PR TITLE
Add Aspect to intercept methods annotated with RabbitHandler

### LINKED ISSUE #13: Implement instrumentation for @RabbitHandler
Add support for @RabbitHandler

e.g. 

```
@Component
@RabbitListener(bindings = {
		@QueueBinding(
				value = @Queue(value = "QUEUE", autoDelete = "true"), 
				exchange = @Exchange(value = "EXCHANGE", autoDelete = "true", type = "direct"), 
				key = "KEY")}
)
public class EventHandler {

	@RabbitHandler
	public void onEvent(MyEvent event) {
               // do something with event
	}

}
```

### PR DESCRIPTION
**Pull request by imod**
This PR adds for methods annotated with `RabbitHandler`, which have have not been instrumented . See #13

Although this does only fix #13, it could be integrated on its own. However, to get it work together with [`Jackson2JsonMessageConverter`](https://docs.spring.io/spring-amqp/docs/2.0.1.RELEASE/reference/html/_reference.html#json-message-converter)(#15) it needs some more work.

So far I have no idea how to fix the #15, what I did as a dirty workaround on for us, was this:

1. I created a wrapper to wrap the `MessageConverter`:
```
public class AmqpMessagingSpanSleuthExtractorMessageConverterWrapper implements MessageConverter {

	private static final Logger LOGGER = LoggerFactory.getLogger(AmqpMessagingSpanSleuthExtractorMessageConverterWrapper.class);

	private final MessageConverter wrappedConverter;
	private final AmqpMessagingSpanManager spanManager;

	public AmqpMessagingSpanSleuthExtractorMessageConverterWrapper(MessageConverter wrappedConverter, AmqpMessagingSpanManager spanManager) {
		this.wrappedConverter = wrappedConverter;
		this.spanManager = spanManager;
	}

	@Override
	public Message toMessage(Object object, MessageProperties messageProperties) throws MessageConversionException {
		return wrappedConverter.toMessage(object, messageProperties);
	}

	@Override
	public Object fromMessage(Message message) throws MessageConversionException {
		try {
			spanManager.beforeHandle(message, new String[] {});
		} catch (Exception e) {
			LOGGER.error("failed to handle sleuth 'span' before converting message!", e);
		}
		return wrappedConverter.fromMessage(message);
	}

}
```

2. I configured the `SimpleRabbitListenerContainerFactory` to use the wrapper instead of the `Jackson2JsonMessageConverter` directly...

```
	@Bean
	public SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory(ConnectionFactory connectionFactory, ObjectMapper mapper, AmqpMessagingSpanManager spanManager) {
		SimpleRabbitListenerContainerFactory factory = new SimpleRabbitListenerContainerFactory();
		factory.setConnectionFactory(connectionFactory);
		factory.setMessageConverter(new AmqpMessagingSpanSleuthExtractorMessageConverterWrapper(jacksonConverter(mapper), spanManager));
		return factory;
	}

	private Jackson2JsonMessageConverter jacksonConverter(ObjectMapper mapper) {
		return new Jackson2JsonMessageConverter(mapper);
	}
```

Additional note about this PR: I'm not sure why `com.netshoes.springframework.cloud.sleuth.instrument.amqp.RabbitListenerAspect#getQueueNames` actually exists, because `com.netshoes.springframework.cloud.sleuth.instrument.amqp.DefaultAmqpMessagingSpanManager#beforeHandle` does not do anything with the passed `queueNames` - it seems this parameter is only there for testing purposes. But I think for this the call is far to expensive and should be removed in `RabbitListenerAspect` and in the new `RabbitHandlerAspect` of this PR.

### CODE REVIEW

**Comment by ignacio83** [comment]
Thanks @imod. I will analyze this PR this week.

File: src/main/java/com/netshoes/springframework/cloud/sleuth/instrument/amqp/RabbitHandlerAspect.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+   * Creates a new instance.
+   *
+   * @param spanManager Span manager for AMQP messaging
+   */
+  public RabbitHandlerAspect(AmqpMessagingSpanManager spanManager) {
+    this.spanManager = spanManager;
+  }
+
+  @Around("@annotation(org.springframework.amqp.rabbit.annotation.RabbitHandler)")
+  public Object executeAroundRabbitHandlerAnnotation(ProceedingJoinPoint call) throws Throwable {
```
Comment (ignacio83) [inline src/main/java/com/netshoes/springframework/cloud/sleuth/instrument/amqp/RabbitHandlerAspect.java]: This code of this method is duplicated from RabbitListenerAspect. What think about refactor that classes creating a new abstract class named AbstractRabbitAspect and move this code for a method handleAroundRabbitAnnotation or something like this?

File: src/test/java/com/netshoes/springframework/cloud/sleuth/instrument/amqp/SpringTestConfiguration.java [hunk 2]
```diff
@@ -1,9 +1,6 @@
 package com.netshoes.springframework.cloud.sleuth.instrument.amqp;
 
-import com.netshoes.springframework.cloud.sleuth.instrument.amqp.mock.AmqpTemplateMockManager;
-import com.netshoes.springframework.cloud.sleuth.instrument.amqp.mock.RabbitListenerMock;
-import com.netshoes.springframework.cloud.sleuth.instrument.amqp.mock.RabbitListenerMockManager;
-import com.netshoes.springframework.cloud.sleuth.instrument.amqp.mock.RabbitTemplateMock;
+import com.netshoes.springframework.cloud.sleuth.instrument.amqp.mock.*;
```
Comment (ignacio83) [inline src/test/java/com/netshoes/springframework/cloud/sleuth/instrument/amqp/SpringTestConfiguration.java]: Please, do not use this form, follow the Google Java Code Style.

**Comment by ignacio83** [comment]
@imod About [`Jackson2JsonMessageConverter`](https://docs.spring.io/spring-amqp/docs/2.0.1.RELEASE/reference/html/_reference.html#json-message-converter) lets continue this discuss in issue #15

File: src/main/java/com/netshoes/springframework/cloud/sleuth/instrument/amqp/RabbitHandlerAspect.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  }
+
+  /**
+   * Get queues names separated with comma from {@link RabbitListener} annotation.
+   *
+   * @param call Call
+   * @return Queues names
+   */
+  private String[] getQueueNames(ProceedingJoinPoint call) {
```
Comment (ignacio83) [inline src/main/java/com/netshoes/springframework/cloud/sleuth/instrument/amqp/RabbitHandlerAspect.java]: You are right about this method. Let's remove.

File: src/test/java/com/netshoes/springframework/cloud/sleuth/instrument/amqp/SpringTestConfiguration.java (hunk 2 above)
Comment (imod) [reply src/test/java/com/netshoes/springframework/cloud/sleuth/instrument/amqp/SpringTestConfiguration.java]: I know this sounds strange, but I do use it and just could not find a way to disable the "star/*" imports :(

**Pull request merged**